services/cd-service/pkg/repository/repository.go

```diff
@@ -719,6 +719,36 @@ func (s *State) GetEnvironmentLocksFromDB(ctx context.Context, transaction *sql.
 	return result, nil
 }
 
+func (s *State) GetApplicationLocksForEnv(ctx context.Context, transaction *sql.Tx, environment types.EnvName) (map[types.AppName]map[string]Lock, error) {
```

Comment:
Please add a span for this function, e.g.:
```suggestion
func (s *State) GetApplicationLocksForEnv(ctx context.Context, transaction *sql.Tx, environment types.EnvName) (map[types.AppName]map[string]Lock, error) {
    span, ctx := StartSpan...
    defer span.Finish(error)
```